src/SecureElementReader.App/Views/Titlebars/WindowsTitleBar.axaml.cs

```diff
@@ -0,0 +1,103 @@
+        private TextBlock systemChromeTitle;
+        private NativeMenuBar seamlessMenuBar;
+        private NativeMenuBar defaultMenuBar;
+
+        public static readonly StyledProperty<bool> IsSeamlessProperty =
+        AvaloniaProperty.Register<WindowsTitleBar, bool>(nameof(IsSeamless));
+
+        public bool IsSeamless
+        {
+            get { return GetValue(IsSeamlessProperty); }
+            set
+            {
+                SetValue(IsSeamlessProperty, value);
+                if (titleBarBackground != null &&
+                    systemChromeTitle != null &&
+                    seamlessMenuBar != null &&
+                    defaultMenuBar != null)
+                {
+                    titleBarBackground.IsVisible = !IsSeamless;
+                    systemChromeTitle.IsVisible = !IsSeamless;
+                    seamlessMenuBar.IsVisible = !IsSeamless;
+                    defaultMenuBar.IsVisible = !IsSeamless;
+
+                    if (!IsSeamless)
+                    {
+                        titleBar.Resources["SystemControlForegroundBaseHighBrush"] = new SolidColorBrush { Color = new Color(255, 0, 0, 0) };
+                    }
+                }
+            }
+        }
+
+        public WindowsTitleBar()
+        {
+            this.InitializeComponent();
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) == false)
+            {
+                this.IsVisible = false;
+            }
+            else
+            {
+                minimizeButton = this.FindControl<Button>("MinimizeButton");
+                closeButton = this.FindControl<Button>("CloseButton");
+                
+
+                minimizeButton.Click += MinimizeWindow;
+                closeButton.Click += CloseWindow;
+                
+
+                titleBar = this.FindControl<DockPanel>("TitleBar");
+                titleBarBackground = this.FindControl<DockPanel>("TitleBarBackground");
+                systemChromeTitle = this.FindControl<TextBlock>("SystemChromeTitle");
+                seamlessMenuBar = this.FindControl<NativeMenuBar>("SeamlessMenuBar");
+                defaultMenuBar = this.FindControl<NativeMenuBar>("DefaultMenuBar");
+
+                SubscribeToWindowState();
+            }
+        }
+
+        private void CloseWindow(object sender, Avalonia.Interactivity.RoutedEventArgs e)
+        {
+            Window hostWindow = (Window)this.VisualRoot;
+            hostWindow.Close();
+        }
+
+        private void MinimizeWindow(object sender, Avalonia.Interactivity.RoutedEventArgs e)
+        {
+            Window hostWindow = (Window)this.VisualRoot;
+            hostWindow.WindowState = WindowState.Minimized;
+        }
+
+        private async void SubscribeToWindowState()
+        {
+            Window hostWindow = (Window)this.VisualRoot;
+
+            while (hostWindow == null)
+            {
+                hostWindow = (Window)this.VisualRoot;
+                await Task.Delay(50);
```

Comment:
What is the purpose of this delay?

Reply:
hostWindow of app is loaded before custom title bar, while loop can't get the information on which OS the app is started so can't select custom title bar. App can't build without this